src/components/pages/ReadArticle.js

```diff
@@ -14,27 +23,79 @@ class ReadArticle extends Component {
   state = {
     Article: {},
     Author: {},
+    AllBoomarked: {},
+    userId: {},
+    isBookmarked: false,
+    isProfileRequested: false,
+    isPreviousBookmarked: false,
+    redirect: false,
   };
 
   componentWillMount() {
     const { slug } = this.props.match.params;
     this.props.getArticle(slug);
+    this.props.getBoomarks(slug);
```

Comment:
@EmyRukundo why does getBookmarks function need a slug as parameter? Additionally, the function signature in src/redux/actions/article.actions.js does not require the slug parameter.

Reply:
thanks, corrected it